### PR TITLE
os-lib 0.10.2, upickle 3.3.1

### LINKED ISSUE #52: update os-lib to 0.10.2
supports os.proc in scala native

@spamegg1 requested

### PR DESCRIPTION
**Pull request by sideeffffect**
fixes #52

### CODE REVIEW

**Comment by bishabosha** [comment]
im not sure how it works, but you also need to generate the changelog files

**Comment by bishabosha** [comment]
ok please can you also run `scala-cli changelog -- --overwrite` which generates the new changelogs

File: changelog/0.4.0/toolkit-test_0.4.0_2.13_changelog.md [hunk 1]
```diff
@@ -33,24 +36,27 @@
...
+         - com.lihaoyi:upickle-core_2.13:3.3.1 (already listed)
      - com.softwaremill.sttp.client4:core_2.13:4.0.0-M14
        - com.softwaremill.sttp.model:core_2.13:1.7.10
        - com.softwaremill.sttp.shared:core_2.13:1.3.18
        - com.softwaremill.sttp.shared:ws_2.13:1.3.18
          - com.softwaremill.sttp.model:core_2.13:1.7.10 (already listed)
          - com.softwaremill.sttp.shared:core_2.13:1.3.18 (already listed)
      - com.softwaremill.sttp.client4:upickle_2.13:4.0.0-M14
-       - com.lihaoyi:upickle_2.13:3.3.0 (already listed)
+       - com.lihaoyi:upickle_2.13:3.3.0
```
Comment (bishabosha) [inline changelog/0.4.0/toolkit-test_0.4.0_2.13_changelog.md:56]: if we update sttp to `4.0.0-M16` then it will come with upickle 3.3.1

Comment (bishabosha) [reply changelog/0.4.0/toolkit-test_0.4.0_2.13_changelog.md:56]: or alternatively `//> using dep com.softwaremill.sttp.client4::upickle::4.0.0-M14,exclude=com.lihaoyi%%upickle` - but probably what I said above is probably better

**Comment by bishabosha** [comment]
@adpi2 are you meant to update the changelogs in a PR like this, or only when we make a new tag?

**Comment by adpi2** [comment]
In any PR that changes the dependencies.

**Comment by bishabosha** [comment]
@adpi2 knows more about this but is on vacation

**Comment by sideeffffect** [comment]
@adpi2 ping :smile_cat:

**Comment by adpi2** [comment]
I'll be back to work tomorrow. Thanks for your patience.

**Review by adpi2 — APPROVED** [review]
I don't know why there was a OOM as I did not manage to reproduce it locally.

But the thing is we needed to update the development version from 0.4.0, which is already released, to 0.5.0. After that change everything is working fine again. I am going to add a `CONTRIBUTING.md` file to clarify that process.

**Comment by sideeffffect** [comment]
Is there a reason why it's not on Maven Central yet?
https://repo1.maven.org/maven2/org/scala-lang/toolkit_native0.5_3/

The job seems to have finished successfully https://github.com/scala/toolkit/actions/runs/10246371483/job/28343314234

**Comment by sideeffffect** [comment]
Actually it failed, yet everything is green :scream_cat: 
```
Uploading files failed!
Possible causes:
- incorrect Sonatype credentials
- incorrect Sonatype server was used, try 'central-s01'
 -> consult publish subcommand documentation
```
https://github.com/scala/toolkit/actions/runs/10246371483/job/28343314234#step:11:120

**Comment by adpi2** [comment]
> Actually it failed, yet everything is green 🙀

Oh thanks for catching that. I think it is an issue with the Sonatype credentials and I am going to try again.

**Comment by lbialy** [comment]
@Gedochao hey, have you seen this? Looks like it's not only mill then.

**Comment by Gedochao** [comment]
> @Gedochao hey, have you seen this? Looks like it's not only mill then.

The error is different, though. It's likely the Sonatype creds issue everyone had last week. Might be sufficient to regenerate the token.

**Comment by adpi2** [comment]
> The error is different, though. It's likely the Sonatype creds issue everyone had last week. Might be sufficient to regenerate the token.

Yes that's what I did. Let's see if it fixes the release.

**Comment by adpi2** [comment]
It's working better now: https://oss.sonatype.org/content/repositories/public/org/scala-lang/toolkit_2.13/0.5.0/

**Comment by sideeffffect** [comment]
https://github.com/com-lihaoyi/mill#0-11-11

**Comment by sideeffffect** [comment]
What bout the fact that failed publish didn't cause the CI to fail/become red?
Is that an issue with the CI setup of this repository? Or an issue with Mill?

**Comment by adpi2** [comment]
> Is that an issue with the CI setup of this repository? Or an issue with Mill?

That's a good question. It could be that we have `continue-on-error: true` in our workflow for some historical reason. But I think we can remove it now.

**Pull request merged**